src/lib/api/acquisition/order.js

```diff
@@ -111,6 +120,7 @@ export const orderApi = {
   create: create,
   update: update,
   list: list,
+  listWithPendingStatus: listWithPendingStatus,
```

Comment:
(minor) I'm in favour of using the request builder for that; say `withQuery` that allows to enter a free text query.

Reply:
It wouldn't work with `withQuery` because we would need to pass arguments in the call here https://github.com/inveniosoftware/react-invenio-app-ils/blob/7574e2d949f18d414100d1dfcecc5b1e4bd6dd86/src/lib/pages/backoffice/DocumentRequest/DocumentRequestDetails/DocumentRequestSteps/ProviderStep/ProviderStep.js#L85
and in `HitsSearch.js` we need to pass only the reference to the function if we want the query string to be taken into account https://github.com/inveniosoftware/react-invenio-app-ils/blob/64754f29ac5dcf7876b4d06095e8cca76379cc72/src/lib/modules/ESSelector/HitsSearch.js#L81

Reply:
Yes, in this case you can always create an anonymous function:
```javascript
query={
  q => orderApi
        .query()
        .withStatus('PENDING')
        .withQuery(q) // Pass the argument
        .qs()
}
```
Or, better create a function in the class and reference it.

Reply:
The query builder will build the query (`queryString` + `SearchCriteria`) but we still need to pass it as an argument in the `list` function to create the url and make the request.

Reply:
Sorry you're right I forgot it returned an url instead of an http request.

Reply:
you could still use the query builder to build the url in borrowingRequest.js instead of hardcoding it